tests/main/cwd/task.yaml

```diff
@@ -9,9 +9,11 @@ details: |
 
 prepare: |
     "$TESTSTOOLS"/snaps-state install-local test-snapd-sh
+    tests.session -u test prepare
```

Comment:
It's not clear whether these changes are related to the snap-confine change; ideally they would live in a prerequisite PR, or, if that's too much trouble, it would help if they lived in a single separate commit.

Reply:
Well, they are obviously. Without session, snap application is not moved to a tracking cgroup and we cannot establish device filtering.